device_doctor/lib/src/device.dart

```diff
@@ -4,22 +4,36 @@
 
 import 'dart:async';
 
+import 'android_device.dart';
 import 'health.dart';
 import 'ios_device.dart';
+import 'utils.dart';
 
 /// Discovers available devices and chooses one to work with.
 abstract class DeviceDiscovery {
```

Comment:
Do we really need this abstract factory when we only support two devices?

Reply:
The abstract class was used to also support `vm` case when no device is there in old cocoon agent. I keep it here to support android and ios. Casey mentioned potential fuchsia device case in the future.
I can refactor to avoid this class if you see any problems.